sycl/source/detail/scheduler/scheduler.hpp

```diff
@@ -745,13 +755,13 @@ class Scheduler {
 
   // Protects stream buffers pool
   std::mutex StreamBuffersPoolMutex;
-  std::map<stream_impl *, StreamBuffers> StreamBuffersPool;
+  std::map<stream_impl *, StreamBuffers *> StreamBuffersPool;
```

Comment:
Why is it a pointer to `StreamBuffers` as map value now?

Reply:
Added comment about this. 
We need to store pointer to the structure with stream buffers because we want to avoid situation when buffers are destructed during destruction of the scheduler. Scheduler is a global object and it can be destructed after all device runtimes are unloaded. destruction of the buffers at this stage will lead to a faliure. In the correct program there will be sync points for all kernels and all allocated resources will be released by the scheduler. If program is not correct and doesn't have necessary sync point then warning will be issued.